### ISSUE TITLE
Failed to validate the SSL certificate for netix.dl.sourceforge.net

### ISSUE THREAD

**Issue by maikenp** [gc3-uzh-ch/elasticluster#472]
Hello!:)

Now with downloading latest version of elasticluster (with the openstack region_name fix) I get a failure here: 

This is on a centos7 machine with python 2.7.5. , trying to set up a cluster with centos6, if that is relevant. 

```
TASK [lmod : Download sources] **************************************************************************************************************************************************************************
fatal: [frontend001]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to validate the SSL certificate for netix.dl.sourceforge.net:443. Make sure your managed systems have a valid CA certificate installed. If the website serving the url uses SNI you need python >= 2.7.9 on your managed machine or you can install the `urllib3`, `pyOpenSSL`, `ndg-httpsclient`, and `pyasn1` python modules to perform SNI verification in python >= 2.6. You can use validate_certs=False if you do not need to confirm the servers identity but this is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/pki/ca-trust/extracted/pem, /etc/pki/tls/certs, /usr/share/ca-certificates/cacert.org, /etc/ansible. The exception msg was: hostname 'netix.dl.sourceforge.net' doesn't match 'mirrors.netix.net'."}
	to retry, use: --limit @/home/centos/elasticluster/src/elasticluster/share/playbooks/site.retry

PLAY RECAP **********************************************************************************************************************************************************************************************
compute001                 : ok=37   changed=22   unreachable=0    failed=0   
frontend001                : ok=58   changed=35   unreachable=0    failed=1   

2017-08-03 13:48:40 elasticluster-test2 gc3.elasticluster[22098] ERROR Command `ansible-playbook /home/centos/elasticluster/src/elasticluster/share/playbooks/site.yml --inventory=/home/centos/.elasticluster/storage/testcluster.inventory --become --become-user=root -v` failed with exit code 2.
2017-08-03 13:48:40 elasticluster-test2 gc3.elasticluster[22098] ERROR Check the output lines above for additional information on this error.
2017-08-03 13:48:40 elasticluster-test2 gc3.elasticluster[22098] ERROR The cluster has likely *not* been configured correctly. You may need to re-run `elasticluster setup` or fix the playbooks.
2017-08-03 13:48:40 elasticluster-test2 gc3.elasticluster[22098] WARNING Cluster `testcluster` not yet configured. Please, re-run `elasticluster setup testcluster` and/or check your configuration

```

**Comment by maikenp** [comment]
And before the failure I got: 

```
TASK [lmod : Install Lmod from the OS repository (Debian-compatible)] ***********************************************************************************************************************************
fatal: [frontend001]: FAILED! => {"changed": false, "failed": true, "msg": "No package matching 'lmod-7.0*' found available, installed or updated", "rc": 126, "results": ["No package matching 'lmod-7.0*' found available, installed or updated"]}
...ignoring
```

**Comment by riccardomurri** [comment]
The second one can be safely ignored -- it's ElastiCluster probing for package existence.

In the first case: I can easily add `validate_certs: no` to avoid the error, but the problem really is that site `netix.dl.sourceforge.net` is using an invalid certificate... Turning off certificate validation kind of defeats using HTTPS in the first place.

I am unsure how to fix this, but it should be OK if you re-run -- the it will pick a different mirror which hopefully has correct HTTPS set up.

**Comment by riccardomurri** [comment]
Can you please try code from branch "issues/472" on http://github.com/riccardomurri/elasticluster ?  It uses a different download URL which avoids SourceForge mirror sites.

**Comment by riccardomurri** [comment]
(Let me know how it goes but note that I am travelling until next Tuesday and be unresponsive.)

**Comment by maikenp** [comment]
Thanks a lot for suggestion.Will try that, and report back. No hurry, have a good travel!

Maiken

On 3 Aug 2017, at 19:33, Riccardo Murri <notifications@github.com<mailto:notifications@github.com>> wrote:


(Let me know how it goes but note that I am travelling until next Tuesday and be unresponsive.)

-

**Comment by maikenp** [comment]
Finally got to try it. Did not work out of the box, get: 

```
TASK [lmod : Install Lmod from the OS repository (Debian-compatible)] **************************************************************************************************************************************
task path: /home/centos/elasticluster_lmod/src/elasticluster/share/playbooks/roles/lmod/tasks/main.yml:9
fatal: [frontend001]: FAILED! => {"changed": false, "failed": true, "msg": "No package matching 'lmod-7.0*' found available, installed or updated", "rc": 126, "results": ["No package matching 'lmod-7.0*' found available, installed or updated"]}
...ignoring

TASK [lmod : Install build-time dependencies (Debian-compatible)] ******************************************************************************************************************************************
task path: /home/centos/elasticluster_lmod/src/elasticluster/share/playbooks/roles/lmod/tasks/build.yml:7
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: {{is_debian_compatible}}

skipping: [frontend001] => (item=gcc)  => {"changed": false, "item": "gcc", "skip_reason": "Conditional result was False", "skipped": true}
skipping: [frontend001] => (item=make)  => {"changed": false, "item": "make", "skip_reason": "Conditional result was False", "skipped": true}
skipping: [frontend001] => (item=libc6-dev)  => {"changed": false, "item": "libc6-dev", "skip_reason": "Conditional result was False", "skipped": true}

TASK [lmod : Install build-time dependencies (RHEL-compatible)] ********************************************************************************************************************************************
task path: /home/centos/elasticluster_lmod/src/elasticluster/share/playbooks/roles/lmod/tasks/build.yml:18
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: {{is_rhel_compatible}}

ok: [frontend001] => (item=@Development tools) => {"changed": false, "item": "@Development tools", "msg": "", "rc": 0, "results": ["@Development tools: Nothing to do"]}

TASK [lmod : Install required dependencies of Lmod (Debian-compatible)] ************************************************************************************************************************************
task path: /home/centos/elasticluster_lmod/src/elasticluster/share/playbooks/roles/lmod/tasks/build.yml:27
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: {{is_debian_compatible}}

skipping: [frontend001] => (item=tcl)  => {"changed": false, "item": "tcl", "skip_reason": "Conditional result was False", "skipped": true}
skipping: [frontend001] => (item=pkg-config)  => {"changed": false, "item": "pkg-config", "skip_reason": "Conditional result was False", "skipped": true}

TASK [lmod : Install required dependencies of Lmod (RHEL-compatible)] **************************************************************************************************************************************
task path: /home/centos/elasticluster_lmod/src/elasticluster/share/playbooks/roles/lmod/tasks/build.yml:37
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: {{is_rhel_compatible}}

ok: [frontend001] => (item=pkgconfig) => {"changed": false, "item": "pkgconfig", "msg": "", "rc": 0, "results": ["pkgconfig-1:0.23-9.1.el6.x86_64 providing pkgconfig is already installed"]}
ok: [frontend001] => (item=tcl) => {"changed": false, "item": "tcl", "msg": "", "rc": 0, "results": ["tcl-1:8.5.7-6.el6.x86_64 providing tcl is already installed"]}

TASK [lmod : Create required directories] ******************************************************************************************************************************************************************
task path: /home/centos/elasticluster_lmod/src/elasticluster/share/playbooks/roles/lmod/tasks/build.yml:56
changed: [frontend001] => (item=/opt/lmod/7.0/) => {"changed": true, "gid": 0, "group": "root", "item": "/opt/lmod/7.0/", "mode": "0755", "owner": "root", "path": "/opt/lmod/7.0/", "secontext": "unconfined_u:object_r:usr_t:s0", "size": 4096, "state": "directory", "uid": 0}
changed: [frontend001] => (item=/opt/lmod/7.0/src/) => {"changed": true, "gid": 0, "group": "root", "item": "/opt/lmod/7.0/src/", "mode": "0755", "owner": "root", "path": "/opt/lmod/7.0/src/", "secontext": "unconfined_u:object_r:usr_t:s0", "size": 4096, "state": "directory", "uid": 0}
ok: [frontend001] => (item=/opt/lmod/7.0/src/) => {"changed": false, "gid": 0, "group": "root", "item": "/opt/lmod/7.0/src/", "mode": "0755", "owner": "root", "path": "/opt/lmod/7.0/src/", "secontext": "unconfined_u:object_r:usr_t:s0", "size": 4096, "state": "directory", "uid": 0}
changed: [frontend001] => (item=/etc/modulefiles/) => {"changed": true, "gid": 0, "group": "root", "item": "/etc/modulefiles/", "mode": "0755", "owner": "root", "path": "/etc/modulefiles/", "secontext": "unconfined_u:object_r:etc_t:s0", "size": 4096, "state": "directory", "uid": 0}

TASK [lmod : Download sources] *****************************************************************************************************************************************************************************
task path: /home/centos/elasticluster_lmod/src/elasticluster/share/playbooks/roles/lmod/tasks/build.yml:68
changed: [frontend001] => {"changed": true, "checksum_dest": null, "checksum_src": "fc1d86678be245698b24fa3d9c126e8267aadf84", "dest": "/opt/lmod/7.0/src/Lmod-7.0.tar.gz", "gid": 0, "group": "root", "md5sum": "568df9cf6f7afeafc10aae47ae17b887", "mode": "0644", "msg": "OK (unknown bytes)", "owner": "root", "secontext": "unconfined_u:object_r:usr_t:s0", "size": 8500322, "src": "/tmp/tmpudu5WK", "state": "file", "status_code": 200, "uid": 0, "url": "https://github.com/TACC/Lmod/archive/7.0.tar.gz"}

TASK [lmod : Extract sources] ******************************************************************************************************************************************************************************
task path: /home/centos/elasticluster_lmod/src/elasticluster/share/playbooks/roles/lmod/tasks/build.yml:85
fatal: [frontend001]: FAILED! => {"changed": true, "cmd": ["tar", "-xzf", "/opt/lmod/7.0/src/7.0.tar.gz", "-C", "/opt/lmod/7.0/src"], "delta": "0:00:00.002458", "end": "2017-08-04 13:47:57.662783", "failed": true, "rc": 2, "start": "2017-08-04 13:47:57.660325", "stderr": "tar (child): /opt/lmod/7.0/src/7.0.tar.gz: Cannot open: No such file or directory\ntar (child): Error is not recoverable: exiting now\ntar: Child returned status 2\ntar: Error is not recoverable: exiting now", "stderr_lines": ["tar (child): /opt/lmod/7.0/src/7.0.tar.gz: Cannot open: No such file or directory", "tar (child): Error is not recoverable: exiting now", "tar: Child returned status 2", "tar: Error is not recoverable: exiting now"], "stdout": "", "stdout_lines": []}
	to retry, use: --limit @/home/centos/elasticluster_lmod/src/elasticluster/share/playbooks/site.retry

PLAY RECAP *************************************************************************************************************************************************************************************************
compute001                 : ok=37   changed=22   unreachable=0    failed=0   
frontend001                : ok=59   changed=36   unreachable=0    failed=1   

2017-08-04 13:47:58 elasticluster-fresh.novalocal gc3.elasticluster[21344] ERROR Command `ansible-playbook /home/centos/elasticluster_lmod/src/elasticluster/share/playbooks/site.yml --inventory=/home/centos/.elasticluster/storage/testcluster.inventory --become --become-user=root -vv` failed with exit code 2.
2017-08-04 13:47:58 elasticluster-fresh.novalocal gc3.elasticluster[21344] ERROR Check the output lines above for additional information on this error.
2017-08-04 13:47:58 elasticluster-fresh.novalocal gc3.elasticluster[21344] ERROR The cluster has likely *not* been configured correctly. You may need to re-run `elasticluster setup` or fix the playbooks.
2017-08-04 13:47:58 elasticluster-fresh.novalocal gc3.elasticluster[21344] WARNING Cluster `testcluster` not yet configured. Please, re-run `elasticluster setup testcluster` and/or check your configuration
```

**Comment by maikenp** [comment]
Seems that in 
`/src/elasticluster/share/playbooks/roles/lmod/tasks/build.yml`
it should say
```
- name: Extract sources
  command:
    'tar -xzf "{{ lmod_build_dir }}/Lmod-{{LMOD_VERSION}}.tar.gz" -C "{{ lmod_build_dir }}"'
  args:
    creates='{{ lmod_build_dir }}/Lmod-{{LMOD_VERSION}}/configure'
```
was missing ´Lmod-´ in the tar command.

**Comment by maikenp** [comment]
Next problem is similar, but now with lua :)

```
TASK [lua : Download Lua sources] ********************************************************************************************************************************************************************************************************************************************************************
task path: /home/centos/elasticluster_lmod/src/elasticluster/share/playbooks/roles/lua/tasks/build.yml:42
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: {{ lua_pkg_install_deb|failed or lua_pkg_install_rpm|failed }}

fatal: [compute001]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to validate the SSL certificate for netix.dl.sourceforge.net:443. Make sure your managed systems have a valid CA certificate installed. If the website serving the url uses SNI you need python >= 2.7.9 on your managed machine or you can install the `urllib3`, `pyOpenSSL`, `ndg-httpsclient`, and `pyasn1` python modules to perform SNI verification in python >= 2.6. You can use validate_certs=False if you do not need to confirm the servers identity but this is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/pki/ca-trust/extracted/pem, /etc/pki/tls/certs, /usr/share/ca-certificates/cacert.org, /etc/ansible. The exception msg was: hostname 'netix.dl.sourceforge.net' doesn't match 'mirrors.netix.net'."}
	to retry, use: --limit @/home/centos/elasticluster_lmod/src/elasticluster/share/playbooks/site.retry


```

**Comment by maikenp** [comment]
Tried fixing it by changing the following in the  ´playbooks/roles/lua/defaults/main.yml´, it seems so far to be ok:

´´´
#LUA_VERSION: '5.1.4.8'
LUA_VERSION: '5.1.4'
#lua_source_url: 'https://sourceforge.net/projects/lmod/files/lua-{{LUA_VERSION}}.tar.gz/download'
lua_source_url: 'https://github.com/LuaDist/lua/archive/{{ LUA_VERSION }}.tar.gz'
´´´

**Comment by riccardomurri** [comment]
hello @maikenp I have pushed my fixes to https://github.com/riccardomurri/elasticluster branch "issues/472" (basically, no more building Lua from sources).  Can you please check that they work?

Note that I did a `git push -f` because I had rebased on "master" so you might need to get a clean copy of the repo or else tell Git to create a new branch.

**Comment by riccardomurri** [comment]
Fixed in commits 2870c98, a2ded3e, and 948b9f5 on the "master" branch.
